### ISSUE TITLE
[BUG]

### ISSUE THREAD

**Issue by gertjana** [stackrox/kube-linter#139]
**System info:**
 - OS: Macos 11.2.1 Big Sur

**Describe the bug**
Kube-linter complains about:
```test-no-readonly-fs.txt: (object: <no namespace>/2048-deployment apps/v1, Kind=Deployment) container "2048" does not have a read-only root file system (check: no-read-only-root-fs, remediation: Set readOnlyRootFilesystem to true in your container's securityContext.)```
while readOnlyRootFilesystem is set to true in the manifest
note that the yaml file is renamed to .txt to allow uploading here
**To Reproduce**
```
> kube-linter lint test-no-readonly-fs.txt
test-no-readonly-fs.yaml: (object: <no namespace>/2048-deployment apps/v1, Kind=Deployment) container "2048" does not have a read-only root file system (check: no-read-only-root-fs, remediation: Set readOnlyRootFilesystem to true in your container's securityContext.)
```
**Sample YAML input**
If applicable, sample YAML input which reproduces the issue.
[test-no-readonly-fs.txt](https://github.com/stackrox/kube-linter/files/6188078/test-no-readonly-fs.txt)

**Expected behavior**
Kube-linter should not warn about this when the suggested remediation is implemented

**Screenshots**

**Additional context**

**Comment by viswajithiii** [comment]
Hi @gertjana, `kube-linter` is actually doing the correct thing in this case, because there is no `readOnlyRootFilesystem` field in the _pod_ security context. (Look at https://k8siuse.com/v1-pod, and look at the list of fields under `spec.securitycontext`.) You need to add this to the security context within the container.

I know this UX isn't ideal. On our roadmap, we have schema validation as an item. After we implement that, it will catch errors like this (that you added the `readOnlyRootFilesystem` in a place that wouldn't be recognized) and hopefully make it less confusing.

**Comment by gertjana** [comment]
Thanks for clarifying, I looked at where the securityContext: runAsNonRoot was supposed to be and assumed the readOnlyRootFilesytem would be at the same level
